libr/io/p/io_zip.c

```diff
@@ -550,28 +551,28 @@ static int r_io_zip_read(RIO *io, RIODesc *fd, ut8 *buf, int count) {
 		return -1;
 	}
 	zfo = fd->data;
-	if (zfo->b->length < io->off) {
-		io->off = zfo->b->length;
+	if (r_buf_size(zfo->b) < io->off) {
+		io->off = r_buf_size(zfo->b);
 	}
 	return r_buf_read_at (zfo->b, io->off, buf, count);
 }
 
 static int r_io_zip_realloc_buf(RIOZipFileObj *zfo, int count) {
 	int res = false;
-	if (count >= 0 && zfo->b->cur + count > zfo->b->length) {
+	if (count >= 0 && r_buf_seek(zfo->b, 0, 1) + count > r_buf_size(zfo->b)) {
 		RBuffer *buffer = r_buf_new ();
 		if (!buffer) {
 			return false;
 		}
-		buffer->buf = malloc (zfo->b->cur + count );
+		buffer->buf = malloc (r_buf_seek(zfo->b, 0, 1) + count );
 		if (!buffer->buf) {
 			r_buf_free (buffer);
 			return false;
 		}
-		buffer->length = zfo->b->cur + count;
-		memcpy (buffer->buf, zfo->b->buf, zfo->b->length);
-		memset (buffer->buf + zfo->b->length, 0, count);
-		buffer->cur = zfo->b->cur;
+		r_buf_resize(buffer, r_buf_seek(zfo->b, 0, 1) + count);
```

Comment:
what about r_buf_get_size() and r_buf_set_size () instead of .size/.resize ?

Reply:
Because you don’t set a size of a buffer, you resize it. Think about a file based buffer... I think resize give more the idea of what the operation does.